modules/nlu/src/backend/engine.ts

```diff
@@ -437,15 +437,14 @@ export default class ScopedEngine implements Engine {
   }
 
   private _setTextWithoutEntities = async (ds: NLUStructure): Promise<NLUStructure> => {
-    ds.sanitizedText = getTextWithoutEntities(ds.entities, ds.rawText).toLowerCase()
+    ds.sanitizedText = getTextWithoutEntities(ds.entities, ds.rawText)
+    ds.sanitizedLowerText = ds.sanitizedText.toLowerCase()
     return ds
   }
 
   private _tokenize = async (ds: NLUStructure): Promise<NLUStructure> => {
-    const text = sanitize(ds.rawText).toLowerCase()
-    ds.lowerText = text
-    const rawTokens = await this.languageProvider.tokenize(text, ds.language)
-    ds.tokens = makeTokenObjects(rawTokens, text)
+    const rawTokens = await this.languageProvider.tokenize(ds.sanitizedLowerText, ds.language)
+    ds.tokens = makeTokens(rawTokens, ds.sanitizedText)
```

Comment:
tokens are created with the ds.sanitizedLowerText so token Objects (makeTokens) should also be made using the same variable:

`
 const rawTokens = await this.languageProvider.tokenize(ds.sanitizedLowerText, ds.language)
 ds.tokens = makeTokens(rawTokens, ds.sanitizedLowerText)

`

Reply:
we want the case in the make tokens